File: src/main/java/org/hw/hw2/Laptop.java
package org.hw.hw2;

public class Laptop extends Product{
    private String model;
    private int ram;
    private String os;
    private String driveType;
    private int driveCapacity;

    public Laptop(String name, String model, int ram, String os,String driveType, int driveCapacity, int price) {
        this.name = name;
        this.model = model;
        this.ram = ram;
        this.os = os;
        this.driveType = driveType;
        this.driveCapacity = driveCapacity;
        this.price = price;
    }

    public String getModel() {
        return model;
    }


    public void setModel(String model) {
        this.model = model;
    }

    public int getRam() {
        return ram;
    }

    public void setRam(int ram) {
        this.ram = ram;
    }

    public String getOs() {
        return os;
    }

    public void setOs(String os) {
        this.os = os;
    }

    public String getDriveType() {
        return driveType;
    }

    public void setDriveType(String driveType) {
        this.driveType = driveType;
    }

    public int getDriveCapacity() {
        return driveCapacity;
    }

    public void setDriveCapacity(int driveCapacity) {
        this.driveCapacity = driveCapacity;
    }

    @Override
    public String toString() {
        return "Ноутбук: " + name +
                ", модель: " + model +
                ", количество опретивной памяти: " + ram +
                ", операционная система: " + os +
                ", тип диска: " + driveType +
                ", объем диска: " + driveCapacity +
                ", цена: " + price;
    }
}

File: src/main/python/hw7/Product/Laptop.py
from hw7.Product.Product import Product


class Laptop(Product):
    name = ''
    model = ''
    ram = 0
    os = ''
    drive_type = ''
    drive_capacity = 0
    price = 0
    
    def __init__(self, name, model, ram, os, drive_type, drive_capacity, price):
        self.name = name
        self.model = model
        self.ram = ram
        self.os = os
        self.drive_type = drive_type
        self.drive_capacity = drive_capacity
        self.price = price

    def get_name(self):
        return self.name

    def set_name(self, name):
        self.name = name

    def get_model(self):
        return self.model

    def set_model(self, model):
        self.model = model

    def get_ram(self):
        return self.ram

    def set_ram(self, ram):
        self.ram = ram

    def get_os(self):
        return self.os

    def set_os(self, os):
        self.os = os

    def get_drive_type(self):
        return self.drive_type

    def set_drive_type(self, drive_type):
        self.drive_type = drive_type

    def get_drive_capacity(self):
        return self.drive_capacity

    def set_drive_capacity(self, drive_capacity):
        self.drive_capacity = drive_capacity

    def get_price(self):
        return self.price

    def set_price(self, price):
        self.price = price

    def __str__(self):
        laptop = ''
        laptop += f'Ноутбук: {self.name}, '
        laptop += f'модель: {self.model}, '
        laptop += f'количество оперативной памяти: {self.ram}, '
        laptop += f'операционная система: {self.os}, '
        laptop += f'тип диска: {self.drive_type}, '
        laptop += f'объем диска: {self.drive_capacity}, '
        laptop += f'цена: {self.price}'
        return laptop
